src/self-managed-debug/src/main.rs

```diff
@@ -0,0 +1,451 @@
+        async fn export_pod_logs(
+            pods: &Api<Pod>,
+            pod_name: &str,
+            file_path: &PathBuf,
+            is_previous: bool,
+        ) -> Result<(), anyhow::Error> {
+            let suffix = if is_previous { "previous" } else { "current" };
+            let file_name = file_path.join(format!("{}.{}.log", pod_name, suffix));
+
+            let logs = pods
+                .logs(
+                    pod_name,
+                    &LogParams {
+                        previous: is_previous,
+                        timestamps: true,
+                        ..Default::default()
+                    },
+                )
+                .await?;
+
+            if logs.is_empty() {
+                eprintln!("No {} logs found for pod {}", suffix, pod_name);
+                return Ok(());
+            }
+
+            let mut file = File::create(&file_name)?;
+            file.write_all(logs.as_bytes())?;
+            println!("Exported {}", file_name.display());
+
+            Ok(())
+        }
+
+        if let Err(e) = export_pod_logs(&pods, &pod_name, &file_path, true).await {
+            let print_error = || {
+                eprintln!(
+                    "Failed to export previous logs for pod {}: {}",
+                    &pod_name, e
+                );
+            };
+
+            if let Some(kube::Error::Api(e)) = e.downcast_ref::<kube::Error>() {
+                if e.code == 400 {
+                    eprintln!("No previous logs available for pod {}", pod_name);
+                } else {
+                    print_error();
+                }
+            } else {
+                print_error();
+            }
+        }
+
+        if let Err(e) = export_pod_logs(&pods, &pod_name, &file_path, false).await {
+            eprintln!("Failed to export current logs for pod {}: {}", &pod_name, e);
+        }
+    }
+    Ok(())
+}
+
+async fn dump_k8s_resources<T>(
+    context: &Context,
+    api: Api<T>,
+    resource_type: &K8sResourceType,
+    namespace: Option<&String>,
+) -> Result<(), anyhow::Error>
+where
+    T: ListableResource,
+    T: Clone,
+    T: std::fmt::Debug,
+    T: serde::Serialize,
+    T: serde::de::DeserializeOwned,
+{
+    let object_list = api.list(&ListParams::default()).await?;
+
+    if object_list.items.is_empty() {
+        let mut err_msg = format!("No {} found", resource_type);
+        if let Some(namespace) = namespace {
+            err_msg = format!("{} for namespace {}", err_msg, namespace);
+        }
+        println!("{}", err_msg);
+        return Ok(());
+    }
+
+    let file_path = format_resource_path(context.start_time, &resource_type.to_string(), namespace);
+    let file_name = file_path.join(format!("{}.yaml", resource_type));
+    create_dir_all(&file_path)?;
+    let mut file = File::create(&file_name)?;
+
+    for item in &object_list.items {
+        serde_yaml::to_writer(&mut file, &item)?;
+    }
```

Comment:
If we're breaking this up per item, we probably want separate files for each item. Alternatively, just serialize the list output directly, rather than looping over items.
Sorry I missed this last time.

```suggestion
    create_dir_all(&file_path)?;

    for item in &object_list.items {
        let file_name = file_path.join(format!("{}.yaml", item.name_unchecked()));
        let mut file = File::create(&file_name)?;
        serde_yaml::to_writer(&mut file, &item)?;
    }
```